### PR TITLE
move fetch webhooks to outer component; fix delete webhook logic; handle some integration uis

### PR DESCRIPTION
**Pull request by Tungnqs**

### CODE REVIEW

File: libs/components/src/lib/components/ClanSettings/Integrations/MainIntegrations/index.tsx [hunk 1]
```diff
@@ -1,30 +1,51 @@
...
+				className={`dark:text-[#b5bac1] text-textLightTheme py-[20px] px-[16px] flex justify-between items-center border dark:border-black border-[#d1d4d9] rounded-md dark:bg-[#2b2d31] bg-bgLightSecondary ${allWebhooks?.length !== 0 ? "cursor-pointer" : ""}`}
+			>
 				<div className="flex gap-3 items-center">
 					<div className="rounded-full p-4 dark:bg-[#111214]">
 						<Icons.WebhooksIcon className="text-[#4e5058] dark:text-[#b5bac1]" />
 					</div>
 					<div>
 						<div className="pb-[3px] font-semibold">Webhooks</div>
-						<div className="text-[12px]">0 webhook</div>
+						<div className="text-[12px]">{allWebhooks?.length} webhooks</div>
```
Comment (thuy1901) [inline libs/components/src/lib/components/ClanSettings/Integrations/MainIntegrations/index.tsx:31]: 0 -> hidden
1 -> 1 webhook
num >=2 -> num webhooks

**Review by thuy1901 — COMMENTED** [review]
ok

**Review by thuy1901 — APPROVED** [review]

File: libs/components/src/lib/components/ClanSettings/Integrations/MainIntegrations/index.tsx [hunk 2]
```diff
@@ -1,30 +1,52 @@
...
 				Customize your server with integrations. Manage webhooks, followed channels and apps, as well as Twitch and YouTube settings for
 				creators. <b className="font-semibold text-[#00a8fc] hover:underline cursor-pointer">Learn more about managing integrations.</b>
 			</div>
 			<div className="border-b-[1px] border-[#616161] my-[32px]"></div>
-			<div className="dark:text-[#b5bac1] text-textLightTheme py-[20px] px-[16px] flex justify-between items-center border dark:border-black border-[#d1d4d9] rounded-md dark:bg-[#2b2d31] bg-bgLightSecondary">
+			<div
+				onClick={() => {
+					if (numberOfWebhooks !== 0) {
+						setIsOpenWebhooks();
+					}
```
Comment (sontung-nccvn) [inline libs/components/src/lib/components/ClanSettings/Integrations/MainIntegrations/index.tsx]: is that okay if numberOfWebhooks == null ?

File: libs/components/src/lib/components/ClanSettings/Integrations/MainIntegrations/index.tsx [hunk 3]
```diff
@@ -1,30 +1,52 @@
...
-						<div className="text-[12px]">0 webhook</div>
+						<div className="text-[12px]">
+							{numberOfWebhooks && numberOfWebhooks > 1 ? numberOfWebhooks + ' webhooks' : numberOfWebhooks + ' webhook'}
+						</div>
 					</div>
 				</div>
-				<div onClick={setIsOpenWebhooks} className="bg-[#5865f2] text-white rounded-md py-2 px-3 cursor-pointer font-semibold">
-					Create Webhook
-				</div>
+				{numberOfWebhooks === 0 ? (
```
Comment (sontung-nccvn) [inline libs/components/src/lib/components/ClanSettings/Integrations/MainIntegrations/index.tsx]: same here, does it need to check for nullish ?

File: libs/components/src/lib/components/ClanSettings/Integrations/index.tsx [hunk 4]
```diff
@@ -24,7 +37,7 @@ const Integrations = () => {
 				)}
 			</h2>
 
-			{!isOpenWebhooks ? <MainIntegrations setIsOpenWebhooks={() => setIsOpenWebhooks(true)} /> : <Webhooks />}
+			{!isOpenWebhooks ? <MainIntegrations allWebhooks={allWebhooks} setIsOpenWebhooks={() => setIsOpenWebhooks(true)} /> : <Webhooks allWebhooks={allWebhooks} parentChannelsInClan={parentChannelsInClan}/>}
```
Comment (sontung-nccvn) [inline libs/components/src/lib/components/ClanSettings/Integrations/index.tsx]: no-negated-condition

File: libs/components/src/lib/components/ClanSettings/Integrations/MainIntegrations/index.tsx (hunk 2 above)
Comment (Tungnqs) [reply libs/components/src/lib/components/ClanSettings/Integrations/MainIntegrations/index.tsx]: resolved

File: libs/components/src/lib/components/ClanSettings/Integrations/MainIntegrations/index.tsx (hunk 3 above)
Comment (Tungnqs) [reply libs/components/src/lib/components/ClanSettings/Integrations/MainIntegrations/index.tsx]: resolved

File: libs/components/src/lib/components/ClanSettings/Integrations/index.tsx (hunk 4 above)
Comment (Tungnqs) [reply libs/components/src/lib/components/ClanSettings/Integrations/index.tsx]: resolved

**Pull request merged**